datalab/mlalpha/_analysis.py

```diff
@@ -10,206 +10,39 @@
-
-    # Load eval results.
-    eval_results_data = p | \
-        beam.Read('ReadEvalResults', beam.io.TextFileSource(eval_results.source,
-            coder=EvalResultsCsvCoder(eval_results)))
-
-    # Join source with results by key.
-    joined_results = {'source': eval_source_data, 'results': eval_results_data} | \
-        beam.CoGroupByKey() | beam.Map('join by key', FeatureSlicingPipeline._join_info)
-
-    feature_metrics_list = []
-    for feature_to_slice in features_to_slice:
-      feature_metrics = joined_results | \
-          beam.Map('slice_get_key_%s' % feature_to_slice,
-                   lambda (k,v),f=feature_to_slice: (v[f], v)) | \
-          beam.CombinePerKey('slice_combine_%s' % feature_to_slice,
-                             AccuracyFn(target_name)) | \
-          beam.Map('slice_prepend_feature_name_%s' % feature_to_slice,
-                   lambda (k,v),f=feature_to_slice: ('%s:%s' % (f, k), v))
-      feature_metrics_list.append(feature_metrics)
-
-    feature_metrics_list | beam.Flatten() | \
-        beam.Map('ToJsonFormat', lambda (k,v): json.dumps({'feature': k, 'metricValues': v})) | \
-        beam.io.WriteToText(output_file, shard_name_template=shard_name_template)
-    return p
-
-
-  def run_local(self, eval_source, eval_results, features_to_slice, metrics, output_file):
-    """Run the pipeline locally. Blocks execution until it finishes.
-
-    Args:
-      eval_source: The only supported format is CsvEvalResults now while we may add more.
-                   Note the source can be either a GCS path or a local path.
-      eval_results: The only supported format is CsvEvalSource now while we may add more.
-                    Note the source can be either a GCS path or a local path.
-      features_to_slice: A list of features to slice on. The features must exist in
-                         eval_source, and can be numeric, categorical, or target.
-      metrics: A list of metrics to compute. For classification, it supports "accuracy",
-               "logloss". For regression, it supports "RMSE".
-      output_file: The path to a local file holding the aggregated results.
-    """
-    p = beam.Pipeline('DirectPipelineRunner')
-    self._pipeline_def(p, eval_source, eval_results, features_to_slice, metrics, output_file,
-                       shard_name_template='')
-    p.run()
-
-
-  def default_pipeline_options(self, output_dir):
-    """Get default DataFlow options. Users can customize it further on top of it and then
-       send the option to run_cloud().
+def csv_to_dataframe(csv_path, schema_path):
+  """Given a CSV file together with its BigQuery schema file in yaml, load
+     content into a dataframe.
 
     Args:
-      output_dir: A GCS path which will be used as base path for tmp and staging dir.
+      csv_path: Input CSV path. Can be local or GCS.
+      schema_path: Input schema path. Can be local or GCS.
```

Comment:
can you give an example of the schema file in the comment? 

After loading the schema file, the order of the column names in the python object is preserved, right?